src/engine/QueryPlanner.cpp

```diff
@@ -790,245 +840,476 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::seedWithScansAndText(
+ParsedQuery::GraphPattern QueryPlanner::seedFromTransitive(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  ParsedQuery::GraphPattern childPlan = seedFromPropertyPath(left, path._children[0], right);
+  ParsedQuery::GraphPattern p;
+  ParsedQuery::GraphPatternOperation* op = new ParsedQuery::GraphPatternOperation(ParsedQuery::GraphPatternOperation::Type::TRANS_PATH);
+  op->_pathData._left = left;
+  op->_pathData._right = right;
+  op->_pathData._childGraphPattern = new ParsedQuery::GraphPattern(std::move(childPlan));
+  p._children.push_back(op);
+  return p;
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromTransitiveMin(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  ParsedQuery::GraphPattern childPlan = seedFromPropertyPath(left, path._children[0], right);
+  ParsedQuery::GraphPattern p;
+  ParsedQuery::GraphPatternOperation* op = new ParsedQuery::GraphPatternOperation(ParsedQuery::GraphPatternOperation::Type::TRANS_PATH);
+  op->_pathData._left = left;
+  op->_pathData._right = right;
+  op->_pathData._min = path._limit;
+  op->_pathData._childGraphPattern = new ParsedQuery::GraphPattern(std::move(childPlan));
+  p._children.push_back(op);
+  return p;
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromTransitiveMax(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  ParsedQuery::GraphPattern childPlan = seedFromPropertyPath(left, path._children[0], right);
+  ParsedQuery::GraphPattern p;
+  ParsedQuery::GraphPatternOperation* op = new ParsedQuery::GraphPatternOperation(ParsedQuery::GraphPatternOperation::Type::TRANS_PATH);
+  op->_pathData._left = left;
+  op->_pathData._right = right;
+  op->_pathData._max = path._limit;
+  op->_pathData._childGraphPattern = new ParsedQuery::GraphPattern(std::move(childPlan));
+  p._children.push_back(op);
+  return p;
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromInverse(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  return seedFromPropertyPath(right, path, left);
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromIri(const std::string& left,
+                                                    const PropertyPath& path,
+                                                    const std::string& right) {
+  ParsedQuery::GraphPattern p;
+  p._whereClauseTriples.push_back(SparqlTriple(left, path, right));
+  return p;
+}
+
+// _____________________________________________________________________________
+std::string QueryPlanner::generateRandomVarName() {
```

Comment:
How do we guarantee these don't collide?

Reply:
Couldn't we just use some (possibly illegal) prefix + a count?

Reply:
Using a fixed instead of a random prefix seems like a good idea. I'd use a colon though, as it is not allowed as the first character of a variable (only [a-z][A-Z][0-9] '_' and some unicode characters are allowed) and it's a less obscure character.